controllers/helmchart_controller.go

```diff
@@ -378,8 +378,17 @@ func (r *HelmChartReconciler) reconcileFromHelmRepository(ctx context.Context,
 			return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 		}
 
+		// Find override file and retrieve contents
+		var valuesData []byte
+		for _, f := range helmChart.Files {
+			if f.Name == chart.Spec.ValuesFile {
```

Comment:
You likely want to use [`filepath.Clean`](https://golang.org/pkg/path/filepath/#Clean) here, as I think the `f.Name` is in the format `some/file.yaml` and you want to match `./some/file.yaml` too.